### PR TITLE
AC: Update accuracy evaluation method for facenet model

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/adapters/reidentification.py [hunk 1]
```diff
@@ -67,6 +67,6 @@ def _extract_predictions(self, outputs_list, meta):
         if len(outputs_list) == 2:
             self.select_output_blob(outputs_list[0])
             emb1, emb2 = outputs_list[0][self.output_blob], outputs_list[1][self.output_blob]
-            return {self.output_blob: emb1 + emb2}
+            return {self.output_blob: np.concatenate((emb1, emb2), axis=1)}
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/reidentification.py:70]: please make embeding joining method configurable (arcface uses sum of embedding, while facenet concat)

File: tools/accuracy_checker/accuracy_checker/metrics/README.md [hunk 2]
```diff
@@ -81,8 +81,13 @@ More detailed information about calculation segmentation metrics you can find [h
   * `uninterpolated_auc` - should area under precision recall curve be computed using trapezoidal rule or directly.
 * `pairwise_accuracy` - pairwise accuracy for object reidentification. Supported representations: `ReIdentificationClassificationAnnotation`, `ReIdentificationPrediction`.
   * `min_score` - min score for determining that objects are different. You can provide value or use `train_median` value which will be calculated if annotations has training subset.
+  * `distance_mode` - allows to choose one of the distance calculation mode (optional).
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/README.md:84]: please provide which values can be selected

File: tools/accuracy_checker/accuracy_checker/metrics/reid.py [hunk 3]
```diff
@@ -231,31 +231,64 @@ def parameters(cls):
                 description="Min score for determining that objects are different. "
                             "You can provide value or use train_median value which will be calculated "
                             "if annotations has training subset."
+            ),
+            'distance_mode': StringField(
+                optional=True, default='0', description='Allows to choose one of the distance calculation mode',
+                choices=['0', '1']
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/reid.py:237]: I think better to use real methods names instead euclidian_distance or cosine distance

File: tools/accuracy_checker/accuracy_checker/metrics/README.md [hunk 4]
```diff
@@ -81,8 +81,13 @@ More detailed information about calculation segmentation metrics you can find [h
...
+  * `subtract_mean` - allows to subtract mean calculated on train embeddings before calculating the distance(optional, default - `False`).
 * `pairwise_accuracy_subsets` - object reidentification pairwise accuracy with division dataset on test and train subsets for calculation mean score. Supported representations: `ReIdentificationClassificationAnnotation`, `ReIdentificationPrediction`.
   * `subset_number` - number of subsets for separating.
+  * `min_score` - min score for determining that objects are different. You can provide value or use `train_median` value which will be calculated if annotations has training subset.
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/README.md:88]: I do not see info about best_train_threshold

File: tools/accuracy_checker/dataset_definitions.yml [hunk 5]
```diff
@@ -541,6 +541,14 @@ datasets:
       - type: pairwise_accuracy_subsets
         subset_number: 2
 
+  - name: lfw_mtcnn_align
+    data_source: lfw_mtcnnpy_160
```
Comment (eaidova) [inline tools/accuracy_checker/dataset_definitions.yml:545]: could you please provide comment how to get this data?

File: tools/accuracy_checker/accuracy_checker/metrics/reid.py [hunk 6]
```diff
@@ -229,33 +229,69 @@ def parameters(cls):
             'min_score': BaseField(
                 optional=True, default='train_median',
                 description="Min score for determining that objects are different. "
-                            "You can provide value or use train_median value which will be calculated "
-                            "if annotations has training subset."
+                            "You can provide value or use train_median or best_train_threshold values "
+                            "which will be calculated if annotations has training subset."
+            ),
+            'distance_mode': StringField(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/reid.py:235]: method is better, more scientific

File: tools/accuracy_checker/accuracy_checker/adapters/reidentification.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/reidentification.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/README.md (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/README.md]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/reid.py (hunk 3 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/reid.py]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/README.md (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/README.md]: Done

File: tools/accuracy_checker/accuracy_checker/metrics/reid.py (hunk 6 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/metrics/reid.py]: Renamed

**Review by eaidova — APPROVED** [review]

**Pull request merged**